File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let minprices=prices[0];
    let maxpro=0;
    for(let i=1;i<prices.length;i++){
        if(minprices>prices[i]){
            minprices=prices[i]
        }
        else{
            let profit=prices[i]-minprices;
            if (profit > maxpro) {
                maxpro = profit;
            }
        }
    }
    return maxpro;
    
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        vector<int> maxi(prices.size(),0);
        maxi[prices.size()-1]=prices[prices.size()-1];
        for(int i=prices.size()-2;i>=0;i--){
            maxi[i]=max(maxi[i+1],prices[i]);
        }
        for(int i=0;i<prices.size();i++){
            maxi[i]=maxi[i]-prices[i];
        }
        sort(maxi.begin(),maxi.end());
        return maxi[prices.size()-1];
        
    }
};